lib/portage/dbapi/vartree.py

```diff
@@ -6247,6 +6252,26 @@ def _quickpkg_dblink(self, backup_dblink, background, logfile):
         finally:
             self.unlockdb()
 
+    def _needs_move(self, mysrc, mydest, mymode, mydmode):
+        """
+        Checks whether the given file at |mysrc| needs to be moved to |mydest| or if
+        they are identical.
+
+        Takes file mode and extended attributes into account.
+        Should only be used for regular files.
+        """
+        if not os.path.exists(mydest):
+            return True
+
+        if mymode != mydmode:
+            return True
+
+        excluded_xattrs = self.settings.get("PORTAGE_XATTR_EXCLUDE", "")
+        if not _cmpxattr(mysrc, mydest, exclude=excluded_xattrs):
+            return True
```

Comment:
We need to make the xattr logic conditional on `"xattr" in self.settings.features`.